lisp/core.slosh

```diff
@@ -250,7 +250,13 @@ Print the documentation for provided symbol.
 
 Section: core
 %#
-(defmacro doc (sym) `(prn (eval (get-prop '~sym :doc-string))))
+(defmacro doc (sym)
+    (let (docs `(get-prop ~sym :doc-string)
+         has-usage (str-starts-with "Usage:" (str-trim (str docs))))
+        (do
+        `(if ~has-usage
```

Comment:
I think this version is fine but it would be a little better to keep this if at compile/macro expand time and produce one or the other of the below prns (i.e. take out the ` here and add one to each prn branch).  This way the logic (that is redundant) to pick the correct prn does not get inserted into the runtime code.

As an aside, should the usage form be modified to always return a 'Usage', either the one in the docstring or the generated one?  I don't have an opinion on that but it seems like a potentially useful tweak?  Actually that may not be viable because the docstring is on the symbol so these questions may be misplaced.

Reply:
taking both suggestions. I like keeping knowledge of the documentation structure in Rust. We know it meets the specifications at runtime off builds from master because the tests pass when we merge things.